### PR TITLE
result of running swiftformat

### PR DESCRIPTION
**Pull request by damfinkel**
## IMPORTANT: DO NOT MERGE

I tested https://github.com/nicklockwood/SwiftFormat to check how it worked. The rules were the default ones.

### CODE REVIEW

File: Core/Extensions/AVFoundation/AVAsset.swift [hunk 1]
```diff
@@ -11,26 +11,26 @@ import ReactiveSwift
 import Result
 
 public extension AVAsset {
-    
+
```
Comment (dtilve) [inline Core/Extensions/AVFoundation/AVAsset.swift]: This is all XC 9 doing right?

**Comment by guidomb** [comment]
At a high level seems OK there are some rules that I would personally change but I think we could start using it to have a consistent style at least inside a same project

File: Core/Extensions/Foundation/Dictionary.swift [hunk 2]
```diff
@@ -9,22 +9,22 @@
...
+
     /**
-        Returns a new dictionary with the contents of self and the given `dictionary`.
-        If both dictionaries have the same key with different value, the result
-        will have that key with the value found in `dictionary`.
-    */
+     Returns a new dictionary with the contents of self and the given `dictionary`.
+     If both dictionaries have the same key with different value, the result
+     will have that key with the value found in `dictionary`.
+     */
```
Comment (dtilve) [inline Core/Extensions/Foundation/Dictionary.swift]: just curious, since my OCD hates this `**` to begin and `*` to end on comments, is there any rationale?

File: Core/Extensions/UIKit/UIAlertController.swift [hunk 3]
```diff
@@ -9,46 +9,44 @@
...
 
 public extension UIAlertController {
-    
+
     /**
      Builds a UIAlertController from a ConfirmationAlertViewModel
-     
+
      - parameter alertViewModel: Models a ConfirmationAlertViewModel
      - seealso: ConfirmationAlertViewModel
```
Comment (dtilve) [inline Core/Extensions/UIKit/UIAlertController.swift]: I know you didn't change anything but since we are touching this file, could you correct the `see also` typos? Unless it's on purpose (there are too many lol).

File: Core/Extensions/UIKit/UIButton.swift [hunk 4]
```diff
@@ -9,39 +9,39 @@
...
+        var attributes: [String: AnyObject] = [NSUnderlineStyleAttributeName: style.rawValue as AnyObject]
         if let colorAttr = color {
             attributes[NSUnderlineColorAttributeName] = colorAttr
         }
         let underlinedText = NSAttributedString(string: title, attributes: attributes)
         setAttributedTitle(underlinedText, for: state)
     }
 
-    /* 
+    /*
```
Comment (dtilve) [inline Core/Extensions/UIKit/UIButton.swift]: Why does this start with `*` and end with `**`. It's the other way round everywhere else in the code right?

File: Core/Extensions/UIKit/UIAlertController.swift (hunk 3 above)
Comment (danielaRiesgo) [reply Core/Extensions/UIKit/UIAlertController.swift]: I think it's on purpose. So that xcode knows that that line should be shown as a see also.

File: Core/Extensions/ReactiveCocoa/Signal.swift [hunk 5]
```diff
@@ -10,53 +10,52 @@ import ReactiveSwift
...
+     which if failed shouldn't finish the whole result signal, stopping new signals
+     from being triggered when a new value arrives at self.
+
+     ```
+     var loginSignal: Signal<(), NoError>
+
+     loginSignal.flatMap(.Latest) { _ -> Signal<MyUser, MyError> in
+     return authService.login()
+     }
+     ```
```
Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/Signal.swift]: Since we are at it, I would add to the doc:
This case could work by doing:
```
loginSignal.flatMap(.Latest) { _ -> Signal<Result<MyUser, MyError>, NoError> in return authService.login().toResultSignal() }
```

File: Core/Extensions/ReactiveCocoa/Signal.swift [hunk 6]
```diff
@@ -10,53 +10,52 @@ import ReactiveSwift
...
+     Filters stream and only passes through the values that respond
+     to the specific type, as elements of that specific type.
 
-         - returns: A signal with value type T and the same error type.
+     - returns: A signal with value type T and the same error type.
      */
     public func filterType<T>() -> Signal<T, Error> {
-        return filter { $0 is T }.map { $0 as! T }  //swiftlint:disable:this force_cast
-        //Can't restrict T to conform/inherit-from Value
+        return filter { $0 is T }.map { $0 as! T } // swiftlint:disable:this force_cast
```
Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/Signal.swift]: I think the change is correct by only to check.
The swiftlint directive needed to have a space after the `//` or to not have the space? There was a way in which swiftlint ignored the comment.

File: Core/Extensions/ReactiveCocoa/SignalProducer.swift [hunk 7]
```diff
@@ -10,75 +10,73 @@ import ReactiveSwift
...
     /**
-         Transforms a `SignalProducer<Value, Error>` to `SignalProducer<Value, NewError>`
-         This is usually pretty useful when the `flatMap` operator is used and the outer
-         producer has `NoError` error type and the inner one a different type of error.
-         
-         - returns: A signal producer with the same value type but with `NewError` as the error type
+     Transforms a `SignalProducer<Value, Error>` to `SignalProducer<Value, NewError>`
+     This is usually pretty useful when the `flatMap` operator is used and the outer
+     producer has `NoError` error type and the inner one a different type of error.
+
```
Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/SignalProducer.swift]: I think we need to add something here. Since we not only transform the error of the producer, but also ignore the previous errors.
Hmm, I think this should only exist for transforming to NoError, or for transforming NoError to anything. Maybe we could make the restrictions for that. Because if you want to transform one error into another, you would use `mapError`.

File: Core/Extensions/ReactiveCocoa/SignalProducer.swift [hunk 8]
```diff
@@ -10,75 +10,73 @@ import ReactiveSwift
...
+     from being triggered when a new value arrives at self.
+
+     For example,
+     ```
+     var myProperty: MutableProperty<CLLocation>
+
+     myProperty.producer.flatMap(.Latest) { clLocation -> SignalProducer<MyLocation, MyError> in
+     return locationService.fetchLocation(clLocation)
+     }
+     ```
```
Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/SignalProducer.swift]: Here also we should add the transformed version, for me.

File: Core/Extensions/ReactiveCocoa/SignalProducer.swift [hunk 9]
```diff
@@ -10,75 +10,73 @@ import ReactiveSwift
...
+     This is usually useful when the `flatMap` triggers different producers
+     which if failed shouldn't finish the whole result producer, stopping new producers
+     from being triggered when a new value arrives at self.
+
+     For example,
+     ```
+     var myProperty: MutableProperty<CLLocation>
+
+     myProperty.producer.flatMap(.Latest) { clLocation -> SignalProducer<MyLocation, MyError> in
+     return locationService.fetchLocation(clLocation)
```
Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/SignalProducer.swift]: Here also about the indentation.

File: Core/Extensions/ReactiveCocoa/SignalProducer.swift [hunk 10]
```diff
@@ -10,75 +10,73 @@ import ReactiveSwift
...
-    */
+     Filters stream and only passes through the values that respond
+     to the specific type, as elements of that specific type.
+
+     - returns: A signal producer with value type T and the same error type.
+     */
     public func filterType<T>() -> SignalProducer<T, Error> {
-        return filter { $0 is T }.map { $0 as! T }  //swiftlint:disable:this force_cast
-        //Can't restrict T to conform/inherit-from Value
+        return filter { $0 is T }.map { $0 as! T } // swiftlint:disable:this force_cast
```
Comment (danielaRiesgo) [inline Core/Extensions/ReactiveCocoa/SignalProducer.swift]: Same about swiftlint comments.

File: Core/Extensions/UIKit/UIFont.swift [hunk 11]
```diff
@@ -20,15 +20,13 @@ public protocol UIFontProvider {
      - seealso: `UIFont.preferredFont(forTextStyle:)`.
      */
     func appFontName(for style: UIFontTextStyle) -> String
-
```
Comment (danielaRiesgo) [inline Core/Extensions/UIKit/UIFont.swift]: :( I don't like this! If there is an empty line above, it should have an empty line below.
(Same for the ones following)

File: Core/Extensions/UIKit/UIView/UIView.swift [hunk 12]
```diff
@@ -41,13 +41,13 @@ public enum BorderPosition {
...
-        hide or show it,
-        change its color, its alpha or other properties.
-    -warning: You are not suppose to change or use constraints with this view, that was already handled for you.
+ Represents a UIView that is used to create a border to another view.
+
+ You can obtain a BorderView by using `add(top/bottom/left/right:, offsets:, useConstraints:)`
+ functions. But the only changes you are expected to do with it are:
+ hide or show it,
+ change its color, its alpha or other properties.
+ -warning: You are not suppose to change or use constraints with this view, that was already handled for you.
```
Comment (danielaRiesgo) [inline Core/Extensions/UIKit/UIView/UIView.swift]: I think this should be `- warning` with a space, I think it's like that in other docs above.

File: Core/Extensions/UIKit/UIView/UIView.swift [hunk 13]
```diff
@@ -62,23 +62,22 @@ public class BorderView: UIView {
         super.init(frame: frame)
     }
 
-    override public init(frame: CGRect) {
+    public override init(frame _: CGRect) {
         fatalError("You shouldn't create a BorderView this way.")
     }
 
-    required public init(coder: NSCoder) {
+    public required init(coder _: NSCoder) {
```
Comment (danielaRiesgo) [inline Core/Extensions/UIKit/UIView/UIView.swift]: Love that there is a rule for this, because we didn't have a convention and it wasn't consistent throughout our files.

File: Core/Utilities/General.swift [hunk 14]
```diff
@@ -9,22 +9,22 @@
 import Foundation
 
 /**
-    Returns the type's simple name.
-    That is to say:
-        without scope prefixes,
-        without properties' values if it's a struct.
+ Returns the type's simple name.
+ That is to say:
+ without scope prefixes,
```
Comment (danielaRiesgo) [inline Core/Utilities/General.swift]: Maybe we should replace the indentation by a bullet? (`- without scope ...`)

File: Core/Utilities/General.swift [hunk 15]
```diff
@@ -9,22 +9,22 @@
...
 /**
-     Returns the concrete type's simple name.
-     That is to say:
-        without scope prefixes,
-        without properties' values if it's a struct.
- 
-    - seealso: type(of:)
+ Returns the concrete type's simple name.
+ That is to say:
+ without scope prefixes,
```
Comment (danielaRiesgo) [inline Core/Utilities/General.swift]: Same here.

File: CoreTests/Extensions/Foundation/StringSpec.swift [hunk 16]
```diff
@@ -12,432 +12,415 @@ import Quick
...
-                let attrs = ["Come and check our interesting posts at Wolox.":
-                                [NSFontAttributeName: "Helvetica-Regular"],
-                             "Wolox":
-                                [NSLinkAttributeName: URL(string: "www.wolox.com.ar")!,
-                                 NSForegroundColorAttributeName: UIColor.blue],
-                             "posts":
-                                [NSForegroundColorAttributeName: UIColor.red],
-                             "and some more":
-                                [NSForegroundColorAttributeName: UIColor.green]]
+                let attrs = [
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/Foundation/StringSpec.swift]: I preferred the other style, but I can stand this one.

File: CoreTests/Extensions/Foundation/StringSpec.swift [hunk 17]
```diff
@@ -12,432 +12,415 @@ import Quick
...
+                        [NSFontAttributeName: "Helvetica-Regular"],
+                    "Wolox":
+                        [
+                            NSLinkAttributeName: URL(string: "www.wolox.com.ar")!,
+                            NSForegroundColorAttributeName: UIColor.blue,
+                        ],
+                    "posts":
+                        [NSForegroundColorAttributeName: UIColor.red],
+                    "and some more":
+                        [NSForegroundColorAttributeName: UIColor.green],
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/Foundation/StringSpec.swift]: Does this work?? With that comma there?

File: CoreTests/Extensions/Foundation/StringSpec.swift [hunk 18]
```diff
@@ -12,432 +12,415 @@ import Quick
...
-                        expect(attributes[NSForegroundColorAttributeName] as? UIColor).to(equal(UIColor.blue))
-                    } else if nsString.substring(with: range) == "posts" {
-                        expect(attributes.count).to(equal(2))
-                        expect(attributes[NSFontAttributeName] as? String).to(equal("Helvetica-Regular"))
-                        expect(attributes[NSForegroundColorAttributeName] as? UIColor).to(equal(UIColor.red))
-                    } else {
-                        fail()
-                    }
+                                               using: { attributes, range, _ in
+                                                   let nsString = string as NSString
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/Foundation/StringSpec.swift]: I do not like this change.

File: CoreTests/Extensions/Foundation/StringSpec.swift [hunk 19]
```diff
@@ -12,432 +12,415 @@ import Quick
...
                 }
-                
             }
-            
         }
-        
     }
-    // swiftlint:enable function_body_length
 
+    // swiftlint:enable function_body_length
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/Foundation/StringSpec.swift]: I preferred it where it was, since it was written because the function ended, not because the "father" function is about to end.

File: CoreTests/Extensions/UIKit/UICollectionViewSpec.swift [hunk 20]
```diff
@@ -12,128 +12,124 @@ import Nimble
...
-                    expect(collectionView.dequeue(footer: NibLoadableCollectionView.self, for: IndexPath(row: 0, section: 0)))
-                        .to(raiseException(named: "NSInternalInconsistencyException"))
-                }
-                
-            }
-            
-        }
-        */
-    }
+         describe("#register(header:) and #dequeue(header:for:)") {
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UICollectionViewSpec.swift]: I would undo this change, since it's code commented, not documentation or just simple comments.

File: CoreTests/Extensions/UIKit/UITableViewSpec.swift [hunk 21]
```diff
@@ -12,156 +12,148 @@ import Nimble
...
-                    let view = tableView.dequeue(footer: NibLoadableTableView.self)
-                    expect(view).to(beNil())
-                }
-                
-            }
-            
-        }
-        */
-        
+         describe("#register(header:) and #dequeue(header:for:)") {
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UITableViewSpec.swift]: Same here.

**Comment by dtilve** [comment]
@damfinkel what should we do about this?

**Comment by danielaRiesgo** [comment]
Can we close this?
